tests_integ/helpers/base_test.py

```diff
@@ -0,0 +1,370 @@
+        cls.lambda_client = boto3.client("lambda")
+        cls.iam_client = boto3.client("iam")
+        cls.api_v2_client = boto3.client("apigatewayv2")
+        cls.sfn_client = boto3.client("stepfunctions")
+
+        if not cls.output_dir.exists():
+            os.mkdir(str(cls.output_dir))
+
+        cls._upload_resources()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._clean_bucket()
+
+    @classmethod
+    def _clean_bucket(cls):
+        """
+        Empties and deletes the bucket used for the tests
+        """
+        s3 = boto3.resource("s3")
+        bucket = s3.Bucket(cls.s3_bucket_name)
+        object_summary_iterator = bucket.objects.all()
+
+        for object_summary in object_summary_iterator:
+            try:
+                cls.s3_client.delete_object(Key=object_summary.key, Bucket=cls.s3_bucket_name)
+            except ClientError as e:
+                LOG.error(
+                    "Unable to delete object %s from bucket %s", object_summary.key, cls.s3_bucket_name, exc_info=e
+                )
+        try:
+            cls.s3_client.delete_bucket(Bucket=cls.s3_bucket_name)
+        except ClientError as e:
+            LOG.error("Unable to delete bucket %s", cls.s3_bucket_name, exc_info=e)
+
+    @classmethod
+    def _upload_resources(cls):
+        """
+        Creates the bucket and uploads the files used by the tests to it
+        """
+        create_bucket(cls.s3_bucket_name, region=cls.my_region)
+
+        current_file_name = ""
+
+        try:
+            for file_name, file_info in FILE_TO_S3_URI_MAP.items():
+                current_file_name = file_name
+                code_path = str(Path(cls.code_dir, file_name))
+                LOG.debug("Uploading file %s to bucket %s", file_name, cls.s3_bucket_name)
+                cls.s3_client.upload_file(code_path, cls.s3_bucket_name, file_name)
+                LOG.debug("File %s uploaded successfully to bucket %s", file_name, cls.s3_bucket_name)
+                file_info["uri"] = cls._get_s3_uri(file_name, file_info["type"])
+        except ClientError as error:
+            LOG.error("Upload of file %s to bucket %s failed", current_file_name, cls.s3_bucket_name, exc_info=error)
+            cls._clean_bucket()
+            raise error
+
+    @classmethod
+    def _get_s3_uri(cls, file_name, uri_type):
+        if uri_type == "s3":
+            return "s3://{}/{}".format(cls.s3_bucket_name, file_name)
+
+        if cls.my_region == "us-east-1":
+            return "https://s3.amazonaws.com/{}/{}".format(cls.s3_bucket_name, file_name)
+        if cls.my_region == "us-iso-east-1":
+            return "https://s3.us-iso-east-1.c2s.ic.gov/{}/{}".format(cls.s3_bucket_name, file_name)
+        if cls.my_region == "us-isob-east-1":
+            return "https://s3.us-isob-east-1.sc2s.sgov.gov/{}/{}".format(cls.s3_bucket_name, file_name)
+
+        return "https://s3-{}.amazonaws.com/{}/{}".format(cls.my_region, cls.s3_bucket_name, file_name)
```

Comment:
Is this correct for China regions? We may want something to determine partition and feed that into the URL determination.

Reply:
Let's discuss this and review the S3 doc, I migrated this from the Java code, it seems some of it might be outdated